### PR TITLE
[CLOUDGA-13718] Get project ID for CLI auth client from an API listAccounts API

### PR DESCRIPTION
**Pull request by harshitha-yb**
listProjects() API requires the permission PROJECT.READ
Api keys with custom roles that do not have PROJECT.READ will not be able to be used with CLI (forbidden errors)
ProjectData has been added to the response of listAccounts() API (which is free from RBAC)
This PR consumes that for the CLI auth client

Test plan:
Tested locally with API keys having custom roles.

P.S. can you also help with the test failures?

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -105,6 +105,10 @@ func (a *AuthApiClient) Ping() ybmclient.ApiGetPingRequest {
 	return a.ApiClient.HealthCheckApi.GetPing(a.ctx)
 }
 
+func (a *AuthApiClient) GetUser() ybmclient.ApiGetUserRequest {
+	return a.ApiClient.UserApi.GetUser(a.ctx)
```
Comment (posriniv) [inline internal/client/client.go:109]: Trying to understand: Where do you want to add GetUser() and also this seems irrelevant to this PR.

**Comment by posriniv** [comment]
@harshitha-yb To fix the UTs, I think you need to change accounts.json to include project data https://github.com/yugabyte/ybm-cli/blob/main/cmd/test/fixtures/account.json. @shinji62 any other thoughts?

**Comment by cdavid** [comment]
We should only merge this PR / release this code only AFTER the backend code reaches PROD.

**Comment by harshitha-yb** [comment]
> @harshitha-yb To fix the UTs, I think you need to change accounts.json to include project data https://github.com/yugabyte/ybm-cli/blob/main/cmd/test/fixtures/account.json. @shinji62 any other thoughts?

Hi! Attempted this but didnt seem to be enough

**Comment by shinji62** [comment]
If `GetProjectID ` is not used anymore then you should remove it.

To fix the UT you need to change the api call you need to change the response here `https://github.com/yugabyte/ybm-cli/blob/main/cmd/cmd_suite_test.go#L73`

File: internal/client/client.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func (a *AuthApiClient) GetProjectID(projectID string) (string, error) {
 		return projectID, nil
 	}
 
-	projectResp, resp, err := a.ListProjects().Execute()
```
Comment (cdavid) [inline internal/client/client.go:140]: Is `ListProjects` still used anywhere else? If not, can you please remove it?

**Review by cdavid — APPROVED** [review]

**Review by cdavid — APPROVED** [review]

**Comment by harshitha-yb** [comment]
> Can we make PROJECT.READ as part of default permissions for all roles or as a requisite permission for CLUSTER.READ or ACCOUNT.READ?

Yeah ideally everyone should have at least ACCOUNT.READ and PROJECT.READ (most paths are /accountId/projectId)
We had had this discussion if you remember- from UI point of view for the "profile" page
For that- instead of giving everyone these permissions we included this info in the /self call
However, we can't access that API with an API key so we had to add this additional project info in the listAccounts API response which is free from RBAC and open to everyone
